pyccel/parser/semantic.py

```diff
@@ -1331,24 +1436,26 @@ def _assign_lhs_variable(self, lhs, d_var, rhs, new_expressions, is_augassign,ar
                                 else:
                                     new_expressions.append(Allocate(a,
                                         shape=a.alloc_shape, order=a.order, status=status))
-                                    # Add memory deallocation for array variables
-                                    self._allocs[-1].append(a)
                             args = new_args
                     else:
                         new_expressions.append(Allocate(lhs, shape=lhs.alloc_shape, order=lhs.order, status=status))
                 # ...
 
                 # ...
-                # Add memory deallocation for array variables
-                if lhs.is_ndarray and not lhs.on_stack:
-                    # Create Deallocate node
-                    self._allocs[-1].append(lhs)
-                # ...
-
-                # Add memory deallocation for class constructor
-                if isinstance(lhs.dtype, CustomDataType) and not lhs.is_alias:
-                    # Create Deallocate node
-                    self._allocs[-1].append(lhs)
+                # Add memory deallocation
```

Comment:
Is it at this level where we should handle not deallocating pointers to arguments?

Reply:
No. In any case, pointers do not deallocate the object that they point at, and anything that is returned is not deallocated.